sound/soc/sof/core.c

```diff
@@ -537,6 +559,8 @@ int snd_sof_device_remove(struct device *dev)
 	if (!IS_ERR_OR_NULL(pdata->pdev_mach))
 		platform_device_unregister(pdata->pdev_mach);
 
+	snd_sof_dsp_reset(sdev);
```

Comment:
I am not comfortable with a remove sequence that's not the mirror of the probe sequence, sorry.